mlrun/run.py

```diff
@@ -1256,716 +1268,161 @@ def wait_for_runs_completion(runs: list, sleep=3, timeout=0, silent=False):
-                self._context = mlrun.get_or_create_ctx("context")
-                break
-
-    def is_context_available(self) -> bool:
-        """
-        Check if a context was found by the method `look_for_context`.
-
-        :returns: True if a context was found and False otherwise.
-        """
-        return self._context is not None
-
-    def parse_inputs(
-        self, args: tuple, kwargs: dict, expected_arguments_types: OrderedDict
-    ) -> tuple:
-        """
-        Parse the given arguments and keyword arguments data items to the expected types.
-
-        :param args:                     The arguments tuple passed to the function.
-        :param kwargs:                   The keyword arguments dictionary passed to the function.
-        :param expected_arguments_types: An ordered dictionary of the expected types of arguments.
-
-        :returns: The parsed args (kwargs are parsed inplace).
-        """
-        # Parse the arguments:
-        parsed_args = []
-        expected_arguments_keys = list(expected_arguments_types.keys())
-        for i, argument in enumerate(args):
-            if (
-                isinstance(argument, mlrun.DataItem)
-                and expected_arguments_types[expected_arguments_keys[i]]
-                != inspect._empty
-            ):
-                parsed_args.append(
-                    self._parse_input(
-                        data_item=argument,
-                        expected_type=expected_arguments_types[
-                            expected_arguments_keys[i]
-                        ],
-                    )
-                )
-                continue
-            parsed_args.append(argument)
-        parsed_args = tuple(parsed_args)  # `args` is expected to be a tuple.
-
-        # Parse the keyword arguments:
-        for key in kwargs.keys():
-            if isinstance(kwargs[key], mlrun.DataItem) and expected_arguments_types[
-                key
-            ] not in [
-                inspect._empty,
-                mlrun.DataItem,
-            ]:
-                kwargs[key] = self._parse_input(
-                    data_item=kwargs[key], expected_type=expected_arguments_types[key]
+    # Check for None value:
+    if log_hint is None:
+        return None
+
+    # If the log hint was provided as a string, construct a dictionary out of it:
+    if isinstance(log_hint, str):
+        # Check if only key is given:
+        if ":" not in log_hint:
+            log_hint = {"key": log_hint}
+        # Check for valid "<key> : <artifact type>" pattern:
+        else:
+            if log_hint.count(":") > 1:
+                raise mlrun.errors.MLRunInvalidArgumentError(
+                    f"Incorrect log hint pattern. Output keys can have only a single ':' in them to specify the "
+                    f"desired artifact type the returned value will be logged as: '<artifact_key> : <artifact_type>', "
```

Comment:
seems like part of the logic here is a dup of `_separate_by_colon` ?

Reply:
you are right and I do use it. This way the other method is only used for inputs.

Reply:
if its dup why not having 1 method for both?

Reply:
its not really a code dup, its just separating by colon and raising error (according to the use case) if two or more colons are there. Same "code" (if you think about it it is 2 lines...) is used in two different sections because they share the pattern of "key : something else". For 2 lines I don't think its necessary to have custom message according to who called the method and put it in utils or something...